### PR TITLE
Add federated_group share type

### PR DESCRIPTION
**Pull request by smesterheide**
In accordance with https://github.com/nextcloud/server/pull/34132

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Thanks!

I wonder how we can document that this type is only available with some version of Nextcloud

**Comment by ChristophWurst** [comment]
I don't think there is a mechanism to check that unfortunately.

What we certainly can do is add a js doc for the property stating what version of Nextcloud this is available.

**Comment by smesterheide** [comment]
I was not sure about the enum and ES5 transpilation, but it seems to work out just fine in my tooling:
grafik

File: lib/index.ts [hunk 1]
```diff
@@ -9,4 +9,6 @@ export enum Type {
     SHARE_TYPE_REMOTE_GROUP = 9,
     SHARE_TYPE_ROOM = 10,
     SHARE_TYPE_DECK = 12,
+    /** New in Nextcloud v26.0.0 */
```
Comment (susnux) [inline lib/index.ts:12]:
```suggestion
    /** @since Nextcloud 26.0.0 */
```

**Review by susnux — APPROVED** [review]

**Pull request merged**